### PR TITLE
[max] Added 'ThingActions' to handle reboot and other Cube actions

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added `ThingActions` to handle reboot and other Cube actions

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by Hilbrand** [comment]
Do you still want to do more testing? And do you want to add the actions invode work-around?

File: bundles/org.openhab.binding.max/README.md [hunk 1]
```diff
@@ -132,15 +130,31 @@ _Cube device configurable parameters_
...
-*   _Restart Cube_ triggers the reboot of a Cube. This can be used if a Cube became unresponsive to commands or no connection can be made. (e.g. if you tried to connect to the Cube with multiple applications at the same time)
+*   `reboot()`: _Restart Cube_ triggers the reboot of a Cube. This can be used if a Cube became unresponsive to commands or no connection can be made. (e.g. if you tried to connect to the Cube with multiple applications at the same time)
 
 On the MAX! devices you can trigger the following action
 
-*   _Delete Device from Cube_ deletes the device from the MAX! Cube. Device will need to be included again! To use this button you need to enable 'Advanced settings'
+*   `deleteFromCube()`: _Delete Device from Cube_ deletes the device from the MAX! Cube. Device will need to be included again!
 
-Note: In Paper UI there are no action buttons.
-You can trigger these actions changing from 'No Action' in the dropdown to the action value.
```
Comment (marcelrv) [inline bundles/org.openhab.binding.max/README.md:146]: This looks like still valid, and the way to trigger the action from the GUI or am I missing something?

**Review by marcelrv — CHANGES_REQUESTED** [review]
Sorry, did not see this PR earlier.
Can you check the readme, as I understand this PR adds the actions for use in rules, but does not remove the actions from the paperUI config, hence would keep that part of the description

**Comment by cweitkamp** [comment]
@marcelrv Thanks for looking into this. My plan was indeed to remove the configuration too. But I first wanted to get an impression if others agree in it too. If you like it I will try to finalize this PR asap.

**Comment by marcelrv** [comment]
my only worry with removing these from the options is that the regular gui currently has no way to trigger these actions.
(when writing the comment earlier I was still assuming there was already something in the UI that allowed it to trigger)

Dunno if that is planned in OH3 new UI, in that case, maybe it is better to remove these actions from the config once we arrive @OH3.0

**Comment by cweitkamp** [comment]
@Hilbrand @marcelrv I incorporated your review comments.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Comment by Hilbrand** [comment]
@marcelrv Can you have a second look?

**Comment by marcelrv** [comment]
Did functional as well, looks working well..... 
Unfortunately there is now some merge conflict to be resolved. 
Outside the merge conflict, LGTM

**Review by marcelrv — CHANGES_REQUESTED** [review]
besides the merge conflict LGTM

**Comment by cweitkamp** [comment]
Thanks for your review. I rebased my PR.

**Review by marcelrv — APPROVED** [review]

**Review by cpmeister — APPROVED** [review]
LGTM

**Pull request merged**